contracts/RockPaperScissors.sol

```diff
@@ -51,72 +54,121 @@ contract RockPaperScissors is Pausable {
         
-        if(games[gameHash].b == address(0)) {
-            // Second player never joined - refund wager
-            safeAddToBalance(msg.sender, games[gameHash].wager);
-        } else if(_aMove == beats[uint(games[gameHash].bMove)]) {
+        Moves bMove = games[gameHash].bMove;
+        if(_aMove == beats[uint(bMove)]) {
             // 'a' wins
-            safeAddToBalance(games[gameHash].a, games[gameHash].wager * 2);
-        } else if (games[gameHash].bMove == beats[uint(_aMove)]) {
+            safeAddToBalance(msg.sender, wager * 2);
+        } else if (bMove == beats[uint(_aMove)]) {
             // 'b' wins
-            safeAddToBalance(games[gameHash].b, games[gameHash].wager * 2);
+            safeAddToBalance(games[gameHash].b, wager * 2);
         } else {
             // draw
-            safeAddToBalance(games[gameHash].a, games[gameHash].wager);
-            safeAddToBalance(games[gameHash].b, games[gameHash].wager);
+            safeAddToBalance(msg.sender, wager);
+            safeAddToBalance(games[gameHash].b, wager);
         }
-        games[gameHash].completed = true;
-        emit LogEndGame(msg.sender, gameHash, _aMove);
+
+        zeroOutGame(gameHash);
+        emit LogEndedGame(msg.sender, gameHash, _aMove);
 
         // TODO: implement incentive for player A to end game even if losing?
         // TODO: end other games with same password?
     }
 
-    function forfeit(bytes32 _gameHash) public {
+    function cancelGame(bytes32 _salt, Moves _aMove) public {
+        bytes32 gameHash = createHash(_salt, msg.sender, _aMove);
+        uint wager = games[gameHash].wager;
+        address playerB = games[gameHash].b;
+        require(wager > 0, "Game is invalid or already completed");
+        require(!games[gameHash].bMoveSubmitted, "Player B has already submitted move");
+        require(playerB == address(0) || now > games[gameHash].deadline, "Deadline for player B to submit move has not yet passed");
+
+        if(playerB != address(0)) {
+            // Player B already joined (but didn't submit move before deadline) -- refund wager
+            safeAddToBalance(playerB, wager);
+        }
+        safeAddToBalance(msg.sender, wager);
+        zeroOutGame(gameHash);
+        emit LogCancelledGame(msg.sender, gameHash);
+    }
+
+    function forceForfeit(bytes32 _gameHash) public {
+        uint wager = games[_gameHash].wager;
         require(games[_gameHash].b == msg.sender, "You didn't join this game");
-        require(games[_gameHash].completed == false, "Game is already complete");
-        require(games[_gameHash].deadline < now, "Deadline for player A to reveal move has not yet passed");
+        require(games[_gameHash].bMoveSubmitted, "You must submit move first");
+        require(wager > 0, "Game is invalid or already completed");
+        require(now > games[_gameHash].deadline, "Deadline for player A to reveal move has not yet passed");
```

Comment:
Option 1. There will be many situations with variables you might want open to calibration after the contract is deployed. 
Option 2. Doesn't completely resolve the issue. 
Option 3. I incline to this because the owner/admin can adjust the parameters for future games but the rules don't change arbitrarily for games that are in progress.